File: mobile-react/src/features/basic-data/countries/types/country.ts
export type CountryStatus = 'active' | 'archived' | 'all';

export type CountrySearchField =
  | 'all'
  | 'nameAr'
  | 'nameEn'
  | 'alpha2Code'
  | 'alpha3Code'
  | 'phoneCode'
  | 'currencyCode';

export type CountrySearchOperator =
  | 'contains'
  | 'doesNotContain'
  | 'equals'
  | 'doesNotEqual'
  | 'startsWith'
  | 'endsWith';

export type CountrySortColumn =
  | 'nameEn'
  | 'nameAr'
  | 'alpha2Code'
  | 'alpha3Code'
  | 'currencyCode'
  | 'createdOn';

export interface Country {
  id: number;
  nameAr: string;
  nameEn: string;
  alpha2Code: string | null;
  alpha3Code: string | null;
  phoneCode: string | null;
  currencyCode: string | null;
  statesCount: number;
  createdOn: string;
  updatedOn: string | null;
  isDeleted: boolean;
}

export type CountryDetail = Omit<Country, 'statesCount'>;

export interface CountryWithStates extends CountryDetail {
  states: { id: number; nameAr: string; nameEn: string; isDeleted: boolean }[];
}

export interface CountryRequest {
  nameAr: string;
  nameEn: string;
  alpha2Code: string | null;
  alpha3Code: string | null;
  phoneCode: string | null;
  currencyCode: string | null;
}

export interface CountryFilters {
  status: CountryStatus;
}

export interface CountryPageQuery extends CountryFilters {
  pageNumber: number;
  pageSize: number;
  search: string;
  searchField: CountrySearchField;
  searchOperator: CountrySearchOperator;
  sortBy: CountrySortColumn;
  sortDirection: 'asc' | 'desc';
}

export interface BulkArchiveCountriesResponse {
  archivedCount: number;
}

File: api/HrManagementSystem.Domain/GeographicalInformation/Countries/Entities/Country.cs
using HrManagementSystem.Domain.GeographicalInformation.Addresses.Entities;
using HrManagementSystem.Domain.GeographicalInformation.States.Entities;

namespace HrManagementSystem.Domain.GeographicalInformation.Countries.Entities;

public class Country : AuditableEntity
{
    public int Id { get; set; }
    public string NameEn { get; set; } = null!;
    public string NameAr { get; set; } = null!;
    public string? Alpha2Code { get; set; } 
    public string? Alpha3Code { get; set; } 
    public string? PhoneCode { get; set; } 
    public string? CurrencyCode { get; set; }
    public virtual ICollection<State> States { get; set; } =[];
    public virtual ICollection<Address> Addresses { get; set; } = [];
}
